lib/segment/src/types.rs

```diff
@@ -1370,6 +1371,69 @@ pub struct Range<T> {
     pub lte: Option<T>,
 }
 
+pub fn try_parse_datetime(s: &str) -> Option<DateTimePayloadType> {
+    // Attempt to parse the input string in RFC 3339 format
+    if let Ok(datetime) = chrono::DateTime::parse_from_rfc3339(s) {
+        return Some(datetime.into());
+    }
+
+    // Attempt to parse the input string in the specified formats
+    // YYYY-MM-DD'T'HH:MM:SS (without timezone or Z)
+    // YYYY-MM-DD HH:MM:SS
+    // YYY-MM-DD HH:MM
+    // YYYY-MM-DD
+    let datetime =
+        if let Ok(naive) = chrono::NaiveDateTime::parse_from_str(s, "%Y-%m-%d'T'%H:%M:%S") {
+            naive
+        } else if let Ok(naive) = chrono::NaiveDateTime::parse_from_str(s, "%Y-%m-%d %H:%M:%S") {
+            naive
+        } else if let Ok(naive) = chrono::NaiveDateTime::parse_from_str(s, "%Y-%m-%d %H:%M") {
+            naive
+        } else if let Ok(naive) = chrono::NaiveDate::parse_from_str(s, "%Y-%m-%d") {
+            naive.into()
+        } else {
+            // parse error, return none
+            return None;
+        };
+
+    // Convert the parsed NaiveDateTime to a DateTime<Utc>
+    let datetime_utc = datetime.and_utc();
+    Some(datetime_utc)
+}
+
+fn deserialize_datetime_option<'de, D>(
+    s: Option<&str>,
+) -> Result<Option<DateTimePayloadType>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let Some(s) = s else {
+        return Ok(None);
+    };
+    // Attempt to parse the input string in RFC 3339 format
+    let parse_result = try_parse_datetime(s);
+    match parse_result {
+        Some(_) => Ok(parse_result),
+        None => Err(serde::de::Error::custom("Invalid date/time format")),
```

Comment:
Maybe we can make this error a bit more helpful.

It now references RFC 3339. Of course, with these changes we support more, but that would at least point a user in the right direction.

```suggestion
        None => Err(serde::de::Error::custom(format!("'{s}' is not in a supported date/time format, please use RFC 3339"))),
```